### PR TITLE
Separate out Plex specifics from ProgrammingSelector component + more

### PR DESCRIPTION
**Pull request by chrisbenincasa**

### CODE REVIEW

File: web/src/hooks/useRouteName.ts [hunk 1]
```diff
@@ -12,33 +24,31 @@ const namedRoutes: Route[] = [
     name: 'New',
   },
   {
-    matcher:
-      /^\/channels\/[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}\/watch$/g,
+    matcher: channelsPageMatcher('watch'),
```
Comment (markdavella) [inline web/src/hooks/useRouteName.ts:27]: Nice cleanup on all these

File: web/src/pages/library/EditCustomShowPage.tsx [hunk 2]
```diff
@@ -57,10 +62,11 @@ export default function EditCustomShowPage({ isNew }: Props) {
   });
 
   useEffect(() => {
+    console.log(customShow, 'reset');
```
Comment (markdavella) [inline web/src/pages/library/EditCustomShowPage.tsx:65]: nit: just a reminder this is here

**Review by markdavella — APPROVED** [review]
LGTM 🚢

File: web/src/pages/library/EditCustomShowPage.tsx (hunk 2 above)
Comment (chrisbenincasa) [reply web/src/pages/library/EditCustomShowPage.tsx:65]: Will clean these up in a follow-up since i have a buncha other pending changes on my branch now

**Pull request merged**